### PR TITLE
Timeout bug fix and State_Initializing

### PR DESCRIPTION
**Pull request by adamstruck**
Added code to handle 'State_Initializing'. This PR should address the timeout issue we were seeing when TES had to pull the docker image it was going to run. 

I still need to update the schema submodule. Waiting on @kellrott to sync changes to a branch for me to point to. 

I may also need to make changes to address the changes we made to the schema regarding https://github.com/ga4gh/task-execution-schemas/pull/22

### CODE REVIEW

File: src/tes/worker/docker.go [hunk 1]
```diff
@@ -124,23 +124,42 @@ func updateAndTrim(l []byte, v []byte) []byte {
...
 		select {
-		case <-timeout:
-			return nil, fmt.Errorf("Error getting metadata for container: %s", err)
+		case <-ctx.Done():
+			return nil
 		default:
-			switch {
-			case err == nil && metadata.State.Running == true:
-				return &metadata, err
+			metadata, err := dclient.ContainerInspect(context.Background(), dcmd.ContainerName)
```
Comment (buchanae) [inline src/tes/worker/docker.go]: Use the `ctx` instead of creating a new context.

File: src/tes/worker/docker.go [hunk 2]
```diff
@@ -46,15 +46,15 @@ func (dcmd DockerCmd) SetupCommand() (*DockerCmd, error) {
...
 	}
 
-	if dcmd.PortBindings != nil {
-		log.Printf("Docker Port Bindings: %v", dcmd.PortBindings)
-		for i := range dcmd.PortBindings {
-			hostPortNum := int(dcmd.PortBindings[i].HostBinding)
+	if dcmd.Ports != nil {
+		log.Printf("Docker Port Bindings: %v", dcmd.Ports)
+		for i := range dcmd.Ports {
+			hostPortNum := int(dcmd.Ports[i].Host)
```
Comment (buchanae) [inline src/tes/worker/docker.go]: Is it necessary to convert to `int()` for the if statement?

File: src/tes/worker/docker.go [hunk 3]
```diff
@@ -46,15 +46,15 @@ func (dcmd DockerCmd) SetupCommand() (*DockerCmd, error) {
...
 			if hostPortNum <= 1024 && hostPortNum != 0 {
 				return nil, fmt.Errorf("Error cannot use restricted ports")
 			}
 			hostPort := strconv.Itoa(hostPortNum)
-			containerPort := strconv.Itoa(int(dcmd.PortBindings[i].ContainerPort))
+			containerPort := strconv.Itoa(int(dcmd.Ports[i].Container))
```
Comment (buchanae) [inline src/tes/worker/docker.go]: Another option would be `fmt.Sprintf("%d:%d", hostPort, containerPort)

File: src/tes/worker/docker.go (hunk 2 above)
Comment (adamstruck) [inline src/tes/worker/docker.go]: it would be a string otherwise

Comment (buchanae) [inline src/tes/worker/docker.go]: This says int32:  https://github.com/ga4gh/task-execution-schemas/pull/22/files#diff-ab216c9558570b57ae6df38431722c38R52

**Comment by buchanae** [comment]
A couple minor questions/suggestions, but otherwise looks sweet!

**Pull request merged**